components/org.wso2.carbon.identity.incommon.mgt/src/main/java/org/wso2/carbon/identity/incommon/mgt/xmlhandler/XMLDownloader.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.identity.incommon.mgt.xmlhandler;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.w3c.dom.Document;
+import org.wso2.carbon.identity.incommon.mgt.util.FileUtil;
+import org.wso2.carbon.identity.incommon.mgt.util.HelperConstant;
+import org.xml.sax.SAXException;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Result;
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+public class XMLDownloader {
+
+    private static final Log log = LogFactory.getLog(XMLDownloader.class);
+
+    public String doHTTPConditionalGET(String lastModifiedDate, String fileURL) {
+
+        log.info("Metadata xml download in process.");
+
+        String newLastModifiedDate = "";
+        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+        DocumentBuilder docBuilder = null;
+        Document doc = null;
+        Transformer transformer = null;
+        Result output = null;
+        Source input = null;
+
+        try {
+            docBuilder = dbf.newDocumentBuilder();
+            URL url = new URL(fileURL);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+            connection.setRequestProperty(HelperConstant.IF_MODIFIED_SINCE, lastModifiedDate);
+            // Checking the response code, 200 - modified(return the resource)/  340 - not modified
+            if (connection.getResponseCode() == 200) {
```

Comment:
Use the `org.apache.http.HttpStatus` to get the constants as [here](https://github.com/wso2/carbon-identity-framework/blob/master/components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/SelfRegistrationMgtClient.java#L184)